client/state/terms/actions.js

```diff
@@ -35,7 +37,15 @@ export function addTerm( siteId, taxonomy, term ) {
 		} ) );
```

Comment:
I'm not sure how optimistic we're being here currently, but do we want to include this in the transient category to be selected prior to persisting?